components/camel-jpa/src/main/java/org/apache/camel/component/jpa/JpaHelper.java

```diff
@@ -60,6 +62,15 @@ public static EntityManager getTargetEntityManager(
             em = getEntityManagerMap(exchange).get(getKey(entityManagerFactory));
         }
 
+        // then try to reuse one from the current transaction if possible
+        if (em == null && TransactionSynchronizationManager.hasResource(entityManagerFactory)) {
```

Comment:
It's not good to reuse the one from JpaTransactionManager. I just wonder if there is any way to reuse one in current transaction

```java
if (em == null && exchange != null && exhange.isTransacted()) {
    // get em from the transaction context in camel?
}
```

@davsclaus any thoughts here?

Reply:
When the splitter creates sub exchanges it copies from the parent.

To reuse the same transaction manager, then we could

1) Add a constant for this in Exchange (so its in core)
2) Copy over the existing TM in the sub exchange if its transacted (in the splitter) (see link below)
3) There are other EIPs that should do #2 (Multicast, Recipient List, Enricher) but not wiretap

https://github.com/apache/camel/blob/991f9ca7898bfea362821ef115d930fece317230/core/camel-core-processor/src/main/java/org/apache/camel/processor/Splitter.java#L231

Reply:
Thanks @davsclaus - I'm thinking to share an EntityManager in the sub exchanges during the transaction. Can you take a review about the changes in Splitter.java? If it's OK, I will apply the same change to other EIPs.

Reply:
Yes I added some comments see below

Reply:
I add the similar changes to `multicast` and `recipientList` processors. Aslo an unit test has been added to test with a pooling data source. For `enrich`, the default aggreate stragtegy is using the new exchange, so it works with transacted. But if the custom aggregate stragtegy is used, the users have to copy  `JpaConstants.ENTITY_MANAGE` propery from the newExchange by themseleves. I also provide `JpaHelper.copyEntityManagers` method to do this copying.